warehouse/identities.go

```diff
@@ -442,24 +442,19 @@ func (wh *HandleT) populateHistoricIdentities(warehouse model.Warehouse) {
 		}
 		defer whManager.Cleanup()
 
-		schemaHandle := SchemaHandle{
-			warehouse:    job.warehouse,
-			stagingFiles: job.stagingFiles,
-			dbHandle:     job.dbHandle,
-			whSchemaRepo: repo.NewWHSchemas(job.dbHandle),
-		}
-		job.schemaHandle = &schemaHandle
+		job.schemaHandler = schema.NewHandler(
+			job.dbHandle,
+			job.warehouse,
+		)
+		schema.WithConfig(job.schemaHandler, config.Default)
 
-		job.schemaHandle.schemaInWarehouse, job.schemaHandle.unrecognizedSchemaInWarehouse, err = whManager.FetchSchema(job.warehouse)
+		job.schemaHandler.SchemaInWarehouse, job.schemaHandler.UnrecognizedSchemaInWarehouse, err = job.schemaHandler.FetchSchemaFromWarehouse(whManager)
 		if err != nil {
 			pkgLogger.Errorf(`[WH]: Failed fetching schema from warehouse: %v`, err)
 			job.setUploadError(err, model.Aborted)
 			return
 		}
 
-		schemaHandle.SkipDeprecatedColumns(job.schemaHandle.schemaInWarehouse)
-		schemaHandle.SkipDeprecatedColumns(job.schemaHandle.unrecognizedSchemaInWarehouse)
-
```

Comment:
This should not be exposed to the caller whether we are skipping deprecated columns or not. Moved this inside the `FetchSchemaFromWarehouse`